rfcs/0000-switch-postgresql-from-ec2-to-rds.md

```diff
@@ -0,0 +1,190 @@
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+## Motivation
+[Motivation]: #motivation
+
+This change would increase our system's reliability and agility.
+Most importantly, this would free up operator/staff time to focus more on higher value work.
+Given our teams' small size and lack of PostgreSQL expertise, it would also likely improve our services' overall reliability.
+In addition, RDS is much more amenable to automation -- and thus likely to be deployed far more consistently,
+  which reduces our chances of creating unintended environmental differences and related bugs.
+Ultimately, I think the argument for switching to RDS is less "why should we" and more "why _wouldn't_ we?"
+Given appropriate testing and verification before making the switch, I can't think of any reasons not to.
+
+As an end result of this change, we can expect comparable or improved performance, with significantly less operator overhead, and dramatically simpler configuration management.
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+[Amazon's Relational Database Service (RDS)](https://aws.amazon.com/rds/) offerings are basically,
+  "click a button and get a PostgreSQL server" -- they're managed services wherein Amazon runs and mostly manages RDS for you.
+RDS is offered in several different "flavours": PostgreSQL, MySQL, their "Aurora" product, and others.
+These services remove many, though not all, of the hassles of running a database reliably: they manage the OS, patching, and backups.
+That leaves us still on the hook to manage: major upgrades, maintenance processes like vaccuuming, and tuning.
+For a team as small as ours, this counts as a rather significant win:
+  we're freeing our time and energy up to focus on more important things.
+
+One of the most important considerations here is cost: RDS isn't free.
+For details, please see the attached
+  [RFC postgresql-on-rds Addendum: AWS Pricing](https://docs.google.com/spreadsheets/d/1eeMVqy9xdctcJO80M1JGVqd6apUaza-hfKGtIKLhIdE/edit?usp=sharing)
+  spreadsheet, but at a high-level:
+
+* RDS Aurora charges about twice as much for compute.
+* RDS Aurora does not charge for extra storage for each compute instance; all instances within a region share the same storage.
+    * As storage-per-instance is currently project to be our largest and fastest-growing AWS expense,
+      this would provide us with very large savings.
+* RDS Aurora also introduces a new charge of $0.20 per 1M I/O operations (both read & write).
+  With our projected number of DB queries, this will not be a major cost driver.
+    * It could _become_ a major cost driver if our usage wildly exceeds projections.
+      But, in that scenario, having the more reliable managed DB-as-a-service backing us becomes an even bigger concern that justifies the cost.
+
```

Comment:
What if we capped storage growth at 2x by limiting the number of years that BB stored to 10 years of claims data? My reading of the cost spreadsheet, indicates that RDS Postgres would be better choice from a cost perspective. True? I suggest making this growth assumption explicit in the proposal.